.github/workflows/cicd-ofas.yaml

```diff
@@ -0,0 +1,105 @@
+name: CICD
+on:
+  push:
+    branches:
+      - ocean-spark-v2
+  pull_request:
+    branches:
+      - ocean-spark-v2
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: go.mod
+
+      - name: Run go mod tidy
+        run: |
+          go mod tidy
+          if ! git diff --quiet; then
+            echo "Please run 'go mod tidy' and commit the changes."
+            git diff
+            false
+          fi
+
+      - name: Generate code
+        run: |
+          make generate
+          if ! git diff --quiet; then
+            echo "Need to re-run 'make generate' and commit the changes."
```

Comment:
nit: I don't see this target in the make file above ?

Reply:
this comes from base makefile and it's just a copy of what is in `integration.yaml`